azurelinuxagent/common/protocol/wire.py

```diff
@@ -1401,6 +1350,40 @@ class SharedConfig(object):
     def __init__(self, xml_text):
         logger.verbose("Load SharedConfig.xml")
         self.xml_text = xml_text
+        self.rdma_ipv4_addr = None
+        self.rdma_mac_addr = None
+
+    def parse(self):
```

Comment:
Made this function public instead of private because we only need to parse this on demand when setting up RDMA, which only happens once in the daemon handler. No need to parse it unnecessarily everytime in case the SharedConfig changes in any goalstate.

Reply:
Thanks for the change. My intention was to achieve consistency with the rest of the code. I know this is done only once, and parsing later is not currently needed, but I do not think the time needed to parse it is significant. If perf is really a concern let's leave as is, otherwise we could have an inconsistent goal state.

Reply:
Makes sense, I was debating with myself about the same thing and then in the end decided on a compromise to move the parsing logic to SharedConfig class but call it on demand. But I do agree to your point, I'll make it consistent with the remaining classes as parsing is not a perf intensive operation.